examples/experimental/dagster-airlift/dagster_airlift/core/sensor/builder.py

```diff
@@ -77,10 +77,38 @@ def check_keys_for_asset_keys(
 
 
 def build_airflow_polling_sensor_defs(
-    airflow_data: AirflowDefinitionsData,
+    *,
+    mapped_defs: Definitions,
+    airflow_instance: AirflowInstance,
     event_transformer_fn: Optional[DagsterEventTransformerFn],
     minimum_interval_seconds: int = DEFAULT_AIRFLOW_SENSOR_INTERVAL_SECONDS,
 ) -> Definitions:
+    """Builds a sensor that polls Airflow for new dag runs and emits materializations for each asset.
+
+    Intended to be used either when providing a user-defined mapping from assets to Airflow dags and tasks,
+    or when performing additional transformation to the returned `Definitions` object from `build_airflow_mapped_defs`.
+    The provided `mapped_defs` function expects a `Definitions` object which contains assets mapped to airflow dags and tasks.
+    An asset is considered "mapped" to a dag if it contains the `dagster_airlift.constants.STANDALONE_DAG_ID_METADATA_KEY` metadata key.
+    An asset is considered "mapped" to a task if the `dagster_airlift.constants.TASK_MAPPING_METADATA_KEY` metadata key is present,
+    and the unique identifier of (dag_id, task_id) is present in that metadata value.
+    This mapping is set automatically by `dagster_airlift.core.build_defs_from_airflow_instance` and `dagster_airlift.core.build_mapped_airflow_defs`,
+    but can be set manually as well.
+
+    Using the `event_transformer_fn` argument, users can provide a function that transforms the materializations emitted by the sensor.
+    The expected return type of this function is an iterable of `AssetMaterialization`, `AssetObservation`, or `AssetCheckEvaluation` objects.
+    Each object is expected to have a metadata key `dagster_airlift.constants.EFFECTIVE_TIMESTAMP_METADATA_KEY` which is a `dagster.TimestampMetadataValue` set.
+    This allows Dagster to correctly order the materializations in the event stream.
+
+    Args:
+        mapped_defs (Definitions): The `Definitions` object containing assets mapped to Airflow dags and tasks.
```

Comment:
be specific here that this means metadata has been applied.

A "mapped" definitions objects this metadata exists on those definitions, and the sensor depends on their existence to function properly.